src/app/operators/dialogs/csv/csv-config/csv-properties/csv-properties-template.component.html

```diff
@@ -159,14 +159,15 @@
                                      style="text-align: left"></wave-dialog-section-heading>
         <form fxLayout="column" fxLayoutGap="1rem" fxLayoutAlign="space-around center"
               style="width: 100%; margin-bottom: 1rem;" [formGroup]="temporalProperties">
-            <mat-slide-toggle formControlName="isTime" flex>The file includes temporal data.</mat-slide-toggle>
+            <mat-slide-toggle id="timeToggle" formControlName="isTime" flex>The file includes temporal data.</mat-slide-toggle>
             <mat-form-field>
-                <mat-select fxFlex
+                <mat-select id="intervalTypeSelect"
+                            fxFlex
                             placeholder="Interval type"
                             formControlName="intervalType"
                 >
-                    <mat-option *ngFor="let intervalType of intervalTypes" [value]="intervalType.value"
-                                [disabled]="([intervalTypes[1], intervalTypes[2]].indexOf(intervalType) >= 0) && header.length <= 3">
+                    <mat-option *ngFor="let intervalType of intervalTypes; let i = index" id="{{'intervalType' + i}}" [value]="intervalType.value"
+                                [disabled]="((propertiesService.spatialProperties$ | async).isWkt ? 1 : 2) + intervalType.columns > header.length">
```

Comment:
Looks like it was worth testing it.

However, you should create a function that describes when you disable it. It is more or less not readable by now.

`[disabled]="noAvailableColumnsForTimeSpecification(propertiesService.spatialProperties$ | async)"`
I'm not quite sure if you can eliminate the subscription as parameter, but I guess this is somehow the reason why the options should be disabled.